src/domain/request/requestExecutor/RandomRequestExecutor.ts

```diff
@@ -0,0 +1,43 @@
+import RandomDotOrgDataFetcher from '@core/application/dataFetcher/random/RandomDotOrgDataFetcherAdapter';
+import RandomSgxDataFetcher from '@core/application/dataFetcher/random/RandomSgxDataFetcherAdapter';
+import {LoggerPort} from '@core/domain/common/port';
+import RequestExecutor from '@core/domain/common/port/RequestExecutorPort';
+import FetchDataUseCase from '@core/domain/common/usecase/FetchDataUseCase';
+import Request from '@core/domain/request/Request';
+import Response from '@core/domain/response/Response';
+
+class RandomRequestExecutor implements RequestExecutor {
+    private fetchDataUseCase: FetchDataUseCase;
+    constructor(
+        private readonly sgxEnabled: boolean,
+        private readonly randomDotOrgApiKey: string,
+        private readonly logger: LoggerPort,
+    ) {
+        this.fetchDataUseCase = new FetchDataUseCase(this.chooseDataFetcher(), this.logger);
+    }
+
+    canHandle(contentType: string): boolean {
+        return 'random' === contentType;
```

Comment:
This string should be extracted to some enum.

Reply:
Agreed. This requires changes in many places (biggest offender is RequestUrlParser) . I plan to do that as a part of enhancement task.